internal/schemautil/userconfig/apiconvert/toapi.go

```diff
@@ -270,6 +270,57 @@ func itemToAPI(
 	return res, o, nil
 }
 
+// processManyToOneKeys is a function that processes many to one keys.
+func processManyToOneKeys(res map[string]interface{}, depthIdx int) {
```

Comment:
Why do we need the `depthIdx` ? I see that we are using it conclusively like this `processManyToOneKeys(va, depthIdx+1)`, but I do not see it used anywhere else.

Reply:
ah, thanks for noticing this, looks like I used it in previous implementation